Content.Server/NPC/Systems/NPCRetaliationSystem.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Content.Server.NPC.Components;
+using Content.Shared.CombatMode;
+using Content.Shared.Damage;
+using Content.Shared.Mobs.Components;
+using Robust.Shared.Collections;
+using Robust.Shared.Timing;
+
+namespace Content.Server.NPC.Systems;
+
+public sealed class NPCRetaliationSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly FactionExceptionSystem _factionException = default!;
+    [Dependency] private readonly NpcFactionSystem _npcFaction = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<NPCRetaliationComponent, DamageChangedEvent>(OnDamageChanged);
+        SubscribeLocalEvent<NPCRetaliationComponent, DisarmedEvent>(OnDisarmed);
+    }
+
+    private void OnDamageChanged(EntityUid uid, NPCRetaliationComponent component, DamageChangedEvent args)
+    {
+        if (!args.DamageIncreased)
+            return;
+
+        if (args.Origin is not { } origin)
+            return;
+
+        TryRetaliate(uid, origin, component);
+    }
+
+    private void OnDisarmed(EntityUid uid, NPCRetaliationComponent component, DisarmedEvent args)
+    {
+        TryRetaliate(uid, args.Source, component);
+    }
+
+    public bool TryRetaliate(EntityUid uid, EntityUid target, NPCRetaliationComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return false;
+
+        // don't retaliate against inanimate objects.
```

Comment:
is this explicitly making them ignore borgs which would become good animal wranglers

Reply:
Borgs have mobstate. All mobs, including silicons, have that component.

Reply:
o good